### PR TITLE
Added data-cy for table widget elements

### PR DESCRIPTION
**Pull request by emidhun**
Fixes: #4791

### CODE REVIEW

File: frontend/src/Editor/Components/Table/Table.jsx [hunk 1]
```diff
@@ -602,20 +606,34 @@ export function Table({
...
+                              <input
+                                type="checkbox"
+                                data-cy={`checkbox-coloumn-${String(column.Header).toLowerCase().replace(/\s+/g, '-')}`}
+                                {...column.getToggleHiddenProps()}
+                              />
+                              <span
+                                className="hide-column-name"
+                                data-cy={`options-coloumn-${String(column.Header).toLowerCase().replace(/\s+/g, '-')}`}
+                              >
+                                {' '}
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Components/Table/Table.jsx:634]:
```suggestion
```

File: frontend/src/Editor/Components/Table/Table.jsx [hunk 2]
```diff
@@ -602,20 +606,34 @@ export function Table({
...
+                      data-cy={`dropdown-hide-column`}
                       className={`dropdown-table-column-hide-common ${
                         darkMode ? 'dropdown-table-column-hide-dark-themed' : 'dropdown-table-column-hide'
                       } `}
                     >
                       <div className="dropdown-item">
                         <IndeterminateCheckbox {...getToggleHideAllColumnsProps()} />
-                        <span className="hide-column-name"> Select All</span>
+                        <span className="hide-column-name" data-cy={`options-select-all-coloumn`}>
+                          {' '}
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Components/Table/Table.jsx:617]:
```suggestion
```

**Review by kavinvenkatachalam — CHANGES_REQUESTED** [review]
@emidhun There are few empty strings which can removed. Rest everything looks good

**Comment by emidhun** [comment]
@kavinvenkatachalam Added the mentioned changes.
Please review.

**Review by kavinvenkatachalam — APPROVED** [review]

**Comment by shubh22** [comment]
Task linked: [CU-3149bpv Add data-cy for table widget elements.](https://app.clickup.com/t/3149bpv)

**Pull request merged**